src/librustc_mir/transform/check_consts/validation.rs

```diff
@@ -600,7 +581,8 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
                 };
 
                 if needs_drop {
-                    self.check_op_spanned(ops::LiveDrop, err_span);
+                    // self.check_op_spanned(ops::LiveDrop, err_span);
```

Comment:
so... in order to do the feature gating you suggested, all we need is to change this to be run if the feature gate is not active and the better logic you implemented will run, too, but since it won't cause more errors, it basically works as redundancy?

Reply:
I think so yes. Should we still force `mir_drops_elaborated_and_const_checked` even if the feature gate is not enabled?

Reply:
Running it helps ensure it's equivalent to the old logic, where the old logic doesn't error, but may regress perf to do so. I think it'd be nice to run both, but if you see it's annoying in tests, only force it if the feature gate is enabled.

Reply:
So I'm second guessing whether we actually want to put these changes behind a feature flag.

The `post_drop_elaboration` part is pretty trivial; the only thing I'm worried about implementing wrong is the query forcing part. We need to figure out if it's okay for a const-checking pass to depend on the precision of drop elaboration, but a feature gate just allows us to delay that discussion. It doesn't help resolve it.

Reply:
The reason we should have *some* feature gate is that this does allow stable const fn to do more afaict (well, at least if we consider `if/match` stable :laughing: ) and we should have a test for that, so please add one in case it's not there already.

Reply:
@nikomatsakis, do you want me to nominate this for lang-team or compiler-team discussion? Specifically the part about depending on drop elaboration to determine whether code compiles?